sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Helpers/CSharpInvocationTracker.cs

```diff
@@ -49,17 +49,14 @@ public override InvocationCondition MatchSimpleNames(params MethodSignature[] me
                     context.Model, context.InvokedMethodSymbol, true, methods);
         }
 
-        public override bool FirstParameterIsStringAndIsNotConstant(InvocationContext context)
-        {
-            if (!base.FirstParameterIsString(context))
+        public override InvocationCondition FirstParameterIsConstant() =>
+            (context) =>
             {
-                return false;
-            }
-            var argumentsSyntax = (context.Invocation as InvocationExpressionSyntax)?
-                .ArgumentList?.Arguments.FirstOrDefault();
-
-            return !(argumentsSyntax?.Expression.IsConstant(context.Model) ?? false);
-        }
+                var argumentList = ((InvocationExpressionSyntax)context.Invocation).ArgumentList;
+                return argumentList != null &&
+                    argumentList.Arguments.Count > 0 &&
+                    argumentList.Arguments[0].Expression.RemoveParentheses().IsConstant(context.Model);
```

Comment:
Is this _RemoveParentheses_ required then? If so, we should put it back into _IsConstant_.

Reply:
Now it is required, because we check for SyntaxKind...